### PR TITLE
fix(#7623): Resize ConductorAxis properly

### LINKED ISSUE #7623: Time Conductor Axis does not resize correctly in certain scenario
This is fixed for the plan and gantt chart views.
However, there is some issue with the time conductor axis. If you load Open MCT with a small window size and switch to Fixed timespan mode, then made the window bigger (horizontally), the times in the axis don't update properly:
image

_Originally posted by @shefalijoshi in https://github.com/nasa/openmct/issues/7587#issuecomment-2008187543_

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7623 

### Describe your changes:

- Registers ResizeObserver for the ConductorAxis correctly using composables
- Adds 3 Playwright snapshot tests for the ConductorAxis

e2e Framework Changes:
- Overloads `page.screenshot()` passing in defaults that may apply to most Open MCT snapshot tests (mask timestamps, disable animations). These can be overridden in the test by passing an options object as normal.
- Provides a new fixture `tick()` that is meant to be used in conjunction with `overrideClock`. This allows tests that have paused clock (`shouldAdvanceTime: false`) to be able to manually tick the clock to trigger ui updates, router params updates, etc.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 1]
```diff
@@ -55,20 +57,69 @@ export default {
     }
   },
   emits: ['pan-axis', 'end-pan', 'zoom-axis', 'end-zoom'],
+  setup() {
+    const axisHolder = ref(null);
```
Comment (scottbell) [inline src/plugins/timeConductor/ConductorAxis.vue:61]: this all looks very familiar 😀

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 2]
```diff
@@ -55,20 +57,69 @@ export default {
...
+    };
+  },
   data() {
     return {
       inPanMode: false,
       dragStartX: undefined,
       dragX: undefined,
-      zoomStyle: {}
+      zoomStyle: {},
+      rect: undefined
```
Comment (scottbell) [inline src/plugins/timeConductor/ConductorAxis.vue:79]: picky, take it or leave, but explicitly defining things as undefined seems weird. can we use null here?

**Review by scottbell — APPROVED** [review]
Had a nitpick, but nothing to stop this. Nice work!

Before:

https://github.com/nasa/openmct/assets/9853862/03638957-5f84-4804-9f0b-ba2c99f532ae

After:

https://github.com/nasa/openmct/assets/9853862/418500e7-6241-417e-be30-5d34b9f3ea2a

**Comment by ozyx** [comment]
Wait to merge, have a visual test inc

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 3]
```diff
@@ -55,20 +57,69 @@ export default {
...
     };
   },
   computed: {
     inZoomMode() {
       return !this.inPanMode;
+    },
+    left() {
+      return this.rect.left;
```
Comment (unlikelyzero) [inline src/plugins/timeConductor/ConductorAxis.vue:87]: interesting, all we need to do is pan/drag in cardinal?

Comment (ozyx) [reply src/plugins/timeConductor/ConductorAxis.vue:87]: not sure what you mean?

Comment (unlikelyzero) [reply src/plugins/timeConductor/ConductorAxis.vue:87]: to add the missing testcoverage identified in the codecov warning. We'd need to a test which pans in cardinal directions

Comment (ozyx) [reply src/plugins/timeConductor/ConductorAxis.vue:87]: ah

File: src/plugins/timeConductor/ConductorAxis.vue (hunk 2 above)
Comment (ozyx) [reply src/plugins/timeConductor/ConductorAxis.vue:79]: It is weird! and I hate doing it. But sometimes switching these things to `null` cascades weird issues down the line where we explicitly check for `undefined`. I think we should be safe to update this one, though.

File: e2e/tests/framework/baseFixtures.e2e.spec.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ import { test } from '../../baseFixtures.js';
 
 test.describe('baseFixtures tests', () => {
   //Skip this test for now https://github.com/nasa/openmct/issues/6785
-  test.fixme('Verify that tests fail if console.error is thrown', async ({ page }) => {
+  test('Verify that tests fail if console.error is thrown', async ({ page }) => {
```
Comment (unlikelyzero) [inline e2e/tests/framework/baseFixtures.e2e.spec.js:33]: Rip

File: karma.conf.cjs [hunk 5]
```diff
@@ -106,7 +112,7 @@ module.exports = async (config) => {
     },
     webpack: webpackConfig,
     webpackMiddleware: {
-      stats: 'errors-warnings'
+      stats: 'detailed' // Changed to 'detailed' for more debugging info
```
Comment (unlikelyzero) [inline karma.conf.cjs:115]: Nice

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
This is great! Just two changes
- add framework tests for the tick functionality
- Update the new visual test to use an a11y check in aftereach 
- update the readme to instruct Windows users to use the Linux container snapshot method

File: e2e/tests/framework/baseFixtures.e2e.spec.js (hunk 4 above)
Comment (ozyx) [reply e2e/tests/framework/baseFixtures.e2e.spec.js:33]: F

File: karma.conf.cjs (hunk 5 above)
Comment (ozyx) [reply karma.conf.cjs:115]: yep, we have source maps for it now too

File: e2e/avpFixtures.js [hunk 6]
```diff
@@ -36,27 +36,67 @@
...
+    /**
+     * Override the screenshot function to always mask a given set of locators which will always
+     * show variance across screenshots. Defaults may be overridden by passing in options to the
+     * screenshot function.
+     * @param {import('@playwright/test').PageScreenshotOptions} options - The options for the screenshot.
+     * @returns {Promise<Buffer>} Returns the screenshot as a buffer.
+     */
+    page.screenshot = async function (options = {}) {
+      const mask = [
+        this.getByLabel('Clock Indicator'), // Mask the clock indicator
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/avpFixtures.js:72]: exactly. any passed in options args overrides the current. so in my snapshot tests I added we can control the clock, so I pass in an empty `mask` array to unmask the timestamps

[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/avpFixtures.js:72]: there's probably a better way but we will cross that bridge eventually

File: e2e/baseFixtures.js [hunk 7]
```diff
@@ -111,6 +110,40 @@ const extendedTest = test.extend({
...
+   *
+   * @param {Object} param0
+   * @param {import('@playwright/test').Page} param0.page
+   * @param {import('@playwright/test').Use} param0.use
+   */
+  tick: async ({ page }, use) => {
+    // eslint-disable-next-line func-style
+    const tick = async (milliseconds) => {
+      await page.evaluate((_milliseconds) => {
+        window.__clock.tick(_milliseconds);
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/baseFixtures.js:142]: done and done

File: e2e/tests/visual-a11y/components/timeConductor.visual.spec.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    const mask = [];
+
+    // Take a snapshot for comparison
+    const snapshot = await page.screenshot({
+      mask
+    });
+    expect(snapshot).toMatchSnapshot('time-conductor-realtime.png');
+  });
+  test(
+    'Visual - Time Conductor Axis Resized @a11y @clock @snapshot',
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/tests/visual-a11y/components/timeConductor.visual.spec.js:77]: when I try to it times out. Could have something to do with the frozen clock.

[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/tests/visual-a11y/components/timeConductor.visual.spec.js:77]: I'll stub it out with a comment for now

File: e2e/tests/framework/baseFixtures.e2e.spec.js [hunk 9]
```diff
@@ -52,3 +53,27 @@ test.describe('baseFixtures tests', () => {
...
+
+  test('Can use clockOptions and tick fixtures to control the clock', async ({ page, tick }) => {
+    let time = await page.evaluate(() => new Date().getTime());
+    expect(time).toBe(MISSION_TIME);
+    await tick(1000);
+    time = await page.evaluate(() => new Date().getTime());
+    expect(time).toBe(MISSION_TIME + 1000 * 1);
+    await tick(1000);
+    time = await page.evaluate(() => new Date().getTime());
+    expect(time).toBe(MISSION_TIME + 1000 * 2);
```
Comment (unlikelyzero) [inline e2e/tests/framework/baseFixtures.e2e.spec.js:77]: Awesome

File: e2e/tests/visual-a11y/components/timeConductor.visual.spec.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+      await tick(1000 * 2);
+
+      const mask = [];
+
+      // Take a snapshot for comparison
+      const snapshot = await page.screenshot({
+        mask
+      });
+      expect(snapshot).toMatchSnapshot('time-conductor-axis-resized.png');
+    }
```
Comment (unlikelyzero) [inline e2e/tests/visual-a11y/components/timeConductor.visual.spec.js:114]: Still missing the a11y check

File: e2e/tests/visual-a11y/faultManagement.visual.spec.js [hunk 11]
```diff
@@ -84,6 +84,12 @@ test.describe('Fault Management Visual Tests', () => {
     await shelveFault(page, 1);
     await changeViewTo(page, 'shelved');
 
+    /* cspell:disable-next-line */
+    // Since fault management is heavily dependent on events (bleh), we need to wait for the correct
+    // element counts
+    await expect(page.getByLabel('Select fault:')).toHaveCount(1);
+    await expect(page.getByLabel('Disposition Actions')).toHaveCount(1);
```
Comment (unlikelyzero) [inline e2e/tests/visual-a11y/faultManagement.visual.spec.js:91]: Nice

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Unless I'm missing it on mobile, we're still not executing the accessibility checks

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**